compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -390,13 +397,18 @@ getMatmulOrIGEMMLoweringConfigAndWorkgroupSize(
 
   Value lhs = operands[0];
   Value rhs = operands[1];
-  Value init;
+
+  Value init = operands[2];
   if (scaled) {
     init = operands[4];
-  } else {
-    init = operands[2];
+    assert(
+        llvm::all_of(operands, [](Value a) { return succeeded(getRank(a)); }) &&
+        "All operands must be a shaped type");
```

Comment:
Check for the shaped type directly if this is what the assertion cares about?
```suggestion
        llvm::all_of(operands, [](Value a) { return isa<ShapedType>(a.getType()); }) &&
        "All operands must be a shaped type");
```